protocol-models/src/main/resources/airbyte_protocol/v0/airbyte_protocol.yaml

```diff
@@ -558,6 +558,7 @@ definitions:
       - overwrite
       #- upsert_dedup # TODO chris: SCD Type 1 can be implemented later
       - append_dedup # SCD Type 1 & 2
+      - overwrite_dedup # thanks, resumable full refresh
```

Comment:
not sure I understand why we need this. Destinations already treat `overwrite` as equivalent to `append` (since the overwrite behavior is triggered off of generation_id / minimum_generation_id), and I thought we wanted to eventually just remove the `overwrite` enum value entirely

Reply:
I *think* we need this for the UI to provide the option?

Reply:
if we need to just add a thing to the enum as a hack for faster delivery, that's fine, but that feels like leaking an implementation detail into the protocol? (it'll also make release more difficult, i.e. we need a new platform x connector compatibility check of "does this destination support OVERWRITE_DEDUP")

In principle, the backend needs three orthogonal pieces of information:
* source sync mode (`full refresh` vs `incremental`)
    * this controls whether platform needs to increment generation_id on every sync
* destination sync mode (`append` vs `dedup`)
    * this is passed directly to the destination connector
* destination data retention (`always truncate` vs `always merge`)
    * this controls whether platform sets `min_generation = 0` or `min_gen = generation_id`

and given that, the UI should show a dropdown:
* full refresh append == `source.full_refresh` + `destination.append` + `merge`
* full refresh dedup == `source.full_refresh` + `destination.dedup` + `merge`
    * this is also a sync mode that we could support "for free" (modulo ui/platform work), and is something that our data team has [asked for in the past](https://airbytehq-team.slack.com/archives/C03C4AVJWG4/p1706661474145379?thread_ts=1706311162.824269&cid=C03C4AVJWG4)
* full refresh overwrite == `source.full_refresh` + `destination.append` + `truncate`
    * this is currently its own destination_sync_mode enum value, but it _shouldn't_ be
* full refresh overwrite dedup == `source.full_refresh` + `destination.dedup` + `truncate`
    * this is the proposed new sync mode
* incremental append == `source.incremental` + `destination.append` + `merge`
* incremental dedup == `source.incremental` + `destination.dedup` + `merge`
* (and filter out invalid combinations, i.e. `source.incremental` + `truncate`)

bundling two of those three options into a single enum feels kind of weird